app/os_x11.go

```diff
@@ -619,8 +619,10 @@ func (h *x11EventHandler) handleEvents() bool {
 			redraw = (*C.XExposeEvent)(unsafe.Pointer(xev)).count == 0
 		case C.FocusIn:
 			w.w.Event(key.FocusEvent{Focus: true})
+			w.w.Event(system.FocusEvent{Focus: true})
```

Comment:
Everywhere you send a system.FocusEvent when a key.FocusEvent is also sent. Why is key.FocusEvent not enough?

Reply:
`key.FocusEvent` is going to be consumed by the foremost key handler, which may be deeply nested within application layout code. Wanting to change a window behavior based on the window focus state is a much higher-level concern. One user asked for this focus detection feature in order to build a dmenu-esque application launcher. If the window loses focus, they want to close the window. Doing that by interrogating the state of every widget to see if any of them are focused doesn't seem like the right design.

Reply:
I don't use key.FocusEvent because of the reason mentioned by @whereswaldon. Also, the FocusEvent can work differently from key.FocusEvent. 

I force-push to make the behavior differently. Now, the FocusEvent is only triggered when the window loses focus, instead of the keyboard focus. The word "focus" could be changed to "active", or something else.